src/celengine/body.h

```diff
@@ -306,11 +304,11 @@ class Body
     void getLifespan(double&, double&) const;
 
     Surface* getAlternateSurface(const std::string&) const;
-    void addAlternateSurface(const std::string&, Surface*);
+    void addAlternateSurface(const std::string&, std::unique_ptr<Surface>&&);
     std::vector<std::string>* getAlternateSurfaceNames() const;
 
-    std::vector<Location*>* getLocations() const;
-    void addLocation(Location*);
+    const std::vector<std::unique_ptr<Location>>* getLocations() const;
```

Comment:
maybe `util::array_view` is better suitable here

Reply:
Replaced with a transform view that returns raw pointers rather than smart pointers.